verilog/tools/preprocessor/verilog_preprocessor_test.sh

```diff
@@ -491,5 +491,74 @@ diff --strip-trailing-cr -u "$MY_EXPECT_FILE" "$MY_OUTPUT_FILE" || {
   exit 1
 }
 
+################################################################################
+echo "=== Test multiple-compilation-unit: passing defines"
+
+cat > "$MY_INPUT_FILE" <<EOF
+\`ifdef A
+  A_TRUE
+\`else
+  A_FALSE
+\`endif
+
+\`ifndef B
+  B_FALSE
+\`else
+  B_TRUE
+\`endif
+EOF
+
+
+cat > "$MY_EXPECT_FILE" <<EOF
+(#293: "A_TRUE")
+(#293: "B_FALSE")
+
+EOF
+
+"$preprocessor" multiple-compilation-unit "$MY_INPUT_FILE" +define+A > "$MY_OUTPUT_FILE" 
+
+status="$?"
+
+[[ $status == 0 ]] || {
+  "Expected exit code 0, but got $status"
+  exit 0
+}
+
+diff --strip-trailing-cr -u "$MY_EXPECT_FILE" "$MY_OUTPUT_FILE" || {
+  exit 1
+}
+
+cat > "$MY_EXPECT_FILE" <<EOF
+(#293: "A_TRUE")
+(#293: "B_TRUE")
+
+EOF
+
+"$preprocessor" multiple-compilation-unit "$MY_INPUT_FILE" +define+A=a_value+B=b_value > "$MY_OUTPUT_FILE" 
+
+status="$?"
+
+[[ $status == 0 ]] || {
+  "Expected exit code 0, but got $status"
+  exit 0
+}
+
+diff --strip-trailing-cr -u "$MY_EXPECT_FILE" "$MY_OUTPUT_FILE" || {
+  exit 1
+}
+
+"$preprocessor" multiple-compilation-unit "$MY_INPUT_FILE" +define+A +define+B > "$MY_OUTPUT_FILE" 
+
+status="$?"
+
+[[ $status == 0 ]] || {
+  "Expected exit code 0, but got $status"
+  exit 0
+}
+
+diff --strip-trailing-cr -u "$MY_EXPECT_FILE" "$MY_OUTPUT_FILE" || {
+  exit 1
+}
+
```

Comment:
Would we also already get expanded macros at this point ? So if we have an input file

```
`A
```

and preprocess that with `+define+A=foobar`, would we get an output file
```
foobar
```
?

If so: it should be part of the test.
If not yet: it needs to be a TODO.

Reply:
Yes, it can be expanded, but I currently have the preprocessor tool configured with expand_macros=0.

Reply:
sounds like a command-line flag that wants to be added :)